### PR TITLE
Cross build for Scala 2.12

### PR DESCRIPTION
**Pull request by jeffreyolchovy**
Problem

There exists a need to publish artifacts that are
binary compatible with Scala 2.12.

Solution

Modify sbt build defintion to cross build for 2.12.

### CODE REVIEW

**Comment by jeffreyolchovy** [comment]
In order to upgrade ostrich for 2.12.x (in order to upgrade finagle for 2.12.x).

**Comment by jeffreyolchovy** [comment]
@mosesn All compiles and tests pass locally for me (for all versions of Scala). Since the project is quite old, and the changeset is quite simple, I am guessing that something may need to be changed in the CI env to get the build passing.

**Comment by mosesn** [comment]
Oh geez, I didn't even remember this was still around.  OK, I'll poke some folks internally and see what we should do about this.

File: build.sbt [hunk 1]
```diff
@@ -4,28 +4,43 @@ version := "3.0.2"
 
 organization := "com.twitter"
 
-scalaVersion := "2.10.4"
+scalaVersion := "2.10.6"
 
-crossScalaVersions := Seq("2.10.4", "2.11.4")
+crossScalaVersions := Seq("2.10.6", "2.11.8", "2.12.0-M4")
```
Comment (taylorleese) [inline build.sbt]: @mosesn Can we kill 2.10.x support?

**Comment by jeffreyolchovy** [comment]
updated to drop 2.10 support and made javac source and target congruent with other projects.

**Pull request closed without merging**